File: 26_RemoveDupArray.js


var removeDuplicates = function(nums) {

    let startIndex = 1;

    for(let i = 1; i < nums.length; i++) {
        if (nums[i] !== nums[i - 1]) {
            nums[startIndex] = nums[i]
            startIndex++
        }
    }
    return startIndex;
};

let main = function() {

    let arr = [1,1,1,2,2,2,4,4,5,6,7,7,8,8,9,9,9];
    let result = removeDuplicates(arr)
    console.log(result);
}

main();

File: 26_RemoveDupArray.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution
{
    public:
        int removeDuplicates(vector<int>& nums) {
            int element = nums[0],k = 1;

            for(int i = 1; i < nums.size(); i++) {
                if(element != nums[i]) {
                    nums[k++] = nums[i];
                    element = nums[i];
                }
            }
            return k;
        }
};

int main()
{
    vector<int> nums = {0,0,1,1,1,2,2,3,3,4};

    Solution obj;
    obj.removeDuplicates(nums);
    return 0;
}